src/test/java/edu/kis/powp/jobs2d/DriverCommandVisitorTest.java

```diff
@@ -0,0 +1,49 @@
+package edu.kis.powp.jobs2d;
+
+import edu.kis.powp.jobs2d.command.ComprisingCommandsCounterVisitor;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class DriverCommandVisitorTest implements ActionListener {
+
+    Job2dDriver driver;
+
+    public DriverCommandVisitorTest(Job2dDriver driver)
+    {
+        this.driver = driver;
+    }
+
+    private final static Logger logger = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+
+    @Override
+    public void actionPerformed(ActionEvent actionEvent) {
+        logger.info("Testing Command Visitor");
+        ComprisingCommandsCounterVisitor commandCounterVisitor = new ComprisingCommandsCounterVisitor(this.driver);
+
+        int expectedSetPostitionCalls = 3;
+        int expectedOperateToCalls = 2;
+        int expectedAllCalls = 5;
+
+        List<DriverCommand> driverCommands = new ArrayList<>();
+        driverCommands.add(new SetPositionCommand(-20, -50));
+        driverCommands.add(new OperateToCommand(-20, -50));
+        driverCommands.add(new SetPositionCommand(-20, -40));
+        driverCommands.add(new OperateToCommand(-20, 50));
+        driverCommands.add(new SetPositionCommand(0, -50));
+
+        driverCommands.forEach((c) -> c.accept(commandCounterVisitor));
```

Comment:
Such tests would make much more sense if a single accept would be called on a CompoundCommand object.